src/qz/utils/SystemUtilities.java

```diff
@@ -468,26 +469,43 @@ public static boolean setSystemLookAndFeel() {
      * @return <code>true</code> if the operation is successful
      */
     public static void centerDialog(Dialog dialog, Point position) {
-        if (position == null || position.getX() == 0 || position.getY() == 0) {
```

Comment:
I assumed the `getXY() != 0` check is now redundant. Was there another purpose for these checks?

Reply:
It's not the best logic in the world, but we're actually using `0,0` as a very unlikely magic number per https://github.com/qzind/tray/blob/a07a96b54691a215e9f93aaff4fcee7a0c5ef436/js/qz-tray.js#L233-L236.

Reply:
Despite it's obvious flaws, I would recommend keeping the `0,0` guard, otherwise, the logs will read that it's out of bounds.  One could argue that with the right monitor setup, it may be possible to have the center of the screen at `0,0`, but at the time Brett and I chose this because the likelihood was very low and it allowed useful logging.

In hindsight, we probably should have just sent `{ position: null }` (or better, [not send it at all](https://github.com/qzind/tray/blob/32192244f9d3051abc3b4efd1c9abf9170129674/src/qz/ws/PrintSocketClient.java#L137)) but reverting this decision would need regression testing in both directions for API and app.

Reply:
It may also be the case that the browsers which didn't support this (e.g. IE and the [old Chakra (pre-Chromium) powered Edge](https://en.wikipedia.org/wiki/Chakra_(JScript_engine) and would return a zero as a safeguard.  Unfortunately whatever discussions Brett and I had at that time are lost as they don't appear to be in the source code, Pull Request or bug report.

Reply:
I tested this on Legacy Edge and IE11 and as it turns out, they send null due to a bug in the JavaScript screen calculation.  This is fixed in 7ead436 and now they too will center the dialog on the correct screen.   I also edited the `0, 0` guard back into the code, since the JSON parsing creates a `Point(0, 0)` and then uses `move(...)`, so additional regression testing would be needed if we wanted to eliminate the magic numbers altogether.

https://github.com/qzind/tray/blob/32192244f9d3051abc3b4efd1c9abf9170129674/src/qz/ws/PrintSocketClient.java#L669-L673

Reply:
Leaving the magic numbers 0,0 for now.  We can revisit at a later time if needed.